ddtrace/opentelemetry/_span.py

```diff
@@ -77,9 +73,9 @@ def __init__(
 
         self._ddspan = datadog_span
         if record_exception is not None:
-            self._record_exception = record_exception
+            self._record_exception = record_exception if record_exception is not None else True
         if set_status_on_exception is not None:
-            self._set_status_on_exception = set_status_on_exception
+            self._set_status_on_exception = set_status_on_exception if set_status_on_exception is not None else True
```

Comment:
The otelspan object should be stateless. All fields set on the OtelSpan should reference a value on the underlying spans (ie `_set_status_on_exception` should set a tag/context field on `OtelSpan._ddspan`).  This way a Datadog Span can always be converted to an OTEL Span without losing information. This property ensures the OpenTelemetry and Datadog APIs interoperable. [Here](https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/opentelemetry/_context.py#L62) is one place where we convert Datadog Spans to Otel Spans. 

I will add a regression test to capture this scenario.

Reply:
What I'm understanding from this is that the setting of a tag on the DD span is a functional requirement. If that's the case, I'll change the code to do so without using the core API, since using a span as a "core context" is an implementation detail of the core, and a deprecated one at that.